src/main/java/bleuauction/bleuauction_be/server/orderMenu/service/OrderMenuComponentService.java

```diff
@@ -0,0 +1,107 @@
+package bleuauction.bleuauction_be.server.orderMenu.service;
+
+import bleuauction.bleuauction_be.server.config.annotation.ComponentService;
+import bleuauction.bleuauction_be.server.member.entity.Member;
+import bleuauction.bleuauction_be.server.menu.entity.Menu;
+import bleuauction.bleuauction_be.server.menu.repository.MenuRepository;
+import bleuauction.bleuauction_be.server.order.entity.Order;
+import bleuauction.bleuauction_be.server.order.repository.OrderRepository;
+import bleuauction.bleuauction_be.server.orderMenu.dto.OrderMenuDTO;
+import bleuauction.bleuauction_be.server.orderMenu.entity.OrderMenu;
+import bleuauction.bleuauction_be.server.orderMenu.entity.OrderMenuStatus;
+import bleuauction.bleuauction_be.server.orderMenu.repository.OrderMenuRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@ComponentService
+@Transactional
+@RequiredArgsConstructor
+public class OrderMenuComponentService {
+
+    private final OrderMenuRepository orderMenuRepository;
+    private final OrderRepository orderRepository;
+    private final MenuRepository menuRepository;
+
+    public ResponseEntity<String> addOrderMenuDTO(Member member, Order order, OrderMenuDTO orderMenuDTO) {
+        Menu selectedMenu = menuRepository.findMenusByMenuNo(orderMenuDTO.getMenuNo());
+
+        if (selectedMenu != null) {
+            OrderMenu orderMenu = new OrderMenu();
+            orderMenu.setMemberNo(member);
+            orderMenu.setOrderNo(order);
+            orderMenu.setOrderMenuCount(orderMenuDTO.getOrderMenuCount());
+            orderMenu.setMenuNo(selectedMenu);
+
+            orderMenuRepository.save(orderMenu);
+            log.info("ordermenu/postnew");
+            return ResponseEntity.status(HttpStatus.CREATED).body("OrderMenu created successfully");
+
+        } else {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Selected menu not found");
+        }
+    }
```

Comment:
여기도 사실 했으면 이렇게 하는게 낫지않았을까 싶어여

쉽게 분류가능한걸 먼저 빼버려서, 들여쓰기를 줄여버리게 하는건데 읽기 편해져요.(경험담)


```suggestion
        if (selectedMenu == null) {
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Selected menu not found");            
        } 
        OrderMenu orderMenu = new OrderMenu();
        orderMenu.setMemberNo(member);
        orderMenu.setOrderNo(order);
        orderMenu.setOrderMenuCount(orderMenuDTO.getOrderMenuCount());
        orderMenu.setMenuNo(selectedMenu);

        orderMenuRepository.save(orderMenu);
        log.info("ordermenu/postnew");
        return ResponseEntity.status(HttpStatus.CREATED).body("OrderMenu created successfully");
    }
```

Reply:
와웅 !!!